### ISSUE TITLE
Release packages for Ubuntu 18.04

### ISSUE THREAD

**Issue by SK-CSE** [apache/couchdb#1314]
well the issue is for installing couch db in latest ubuntu 18.04, i am unable to install it in my ubuntu 18.04,
any help would be appreciated. The repository 'https://apache.bintray.com/couchdb-deb' does not have a Release file for ubuntu 18.04

**Comment by alxndrsn** [comment]
@SK-CSE I ran into the same issue.  It may not help in your case, but I worked around it by using [`pouchdb-server`](https://github.com/pouchdb/pouchdb-server) instead.

**Comment by garethbowen** [comment]
@SK-CSE Any update on this?

**Comment by SK-CSE** [comment]
no.. still waiting for release package

**Comment by wohali** [comment]
Packages are coming, please be patient.

pouchdb-server is not an equivalent solution, nor is it anywhere near as performant. (We love the pouch folks!)

**Comment by browndav** [comment]
This is low-priority, but can I ask if you have a good source for couchdb vs. pouchdb benchmark data that covers one or more of the pouchdb backends? I'd like to understand more about the performance characteristics, but don't want to duplicate effort if someone's already covered this ground thoroughly. Thanks in advance.

**Comment by wohali** [comment]
Talk to the developers of pouchdb. Nolan Lawson said he created pouchdb-server as a development tool and that he thinks it'd be terrible in production at any sort of scale.

**Comment by mattpr** [comment]
Just curious... is there something distro specific (besides systemd/upstart/init files) that prevents the provision of a generic linux-x64 binary (say in addition to the distro specific packages)?  Having a generic binary would make it much easier to deploy to new or unsupported distros.  End-users would obviously have to decide if they want to run couchdb as a service and add their own systemd units (or whatever).... but it would be a nice alternative to building from source on unsupported production systems.

**Comment by SK-CSE** [comment]
thanks @wohali :)

**Comment by wohali** [comment]
@mattpr we depend on a large number of libraries and runtimes, so it is better for us not to be shipping ourselves, and having the OS provide. thus packages are one of the better options for distributing couchdb.

the snap package is currently unmaintained but promiseds what you describe. there is also the docker image that many seem to like.

**Comment by browndav** [comment]
I think I'm seeing the same on 16.04 LTS FWIW.

```
W: The repository 'https://apache.bintray.com/couchdb-deb {distribution} Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch https://apache.bintray.com/couchdb-deb/dists/{distribution}/main/binary-amd64/Packages  404  Not Found
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by mattpr** [comment]
@browndav  you need to replace `{distribution}` with the specific distro you want the package for (e.g. "xenial" for 16.04).

http://docs.couchdb.org/en/latest/install/unix.html#installation-using-the-apache-couchdb-convenience-binary-packages

**Comment by wohali** [comment]
The new packages are ready and will be uploaded soon - we are performing final acceptance testing on them.

**Comment by simonmichael** [comment]
These will be much appreciated. I have taken down a busy site by upgrading to 18.04 too soon..

**Comment by simonmichael** [comment]
Because I'm dying here, could I ask you for another ETA @wohali  ? Does soon mean hours, days or weeks ?

**Comment by simonmichael** [comment]
Almost a stopgap fix for me: I can see the successfully built artifacts for bionic at https://builds.apache.org/blue/organizations/jenkins/CouchDB/detail/master/271/artifacts.. but for amd64. Perhaps I should give the centos rpm a try.

**Comment by simonmichael** [comment]
No.. alien says "couchdb-2.2.0.000766c-1.el7.x86_64.rpm is for architecture amd64 ; the package cannot be built on this system".

**Comment by wohali** [comment]
@simonmichael those packages are not for general use, and are definitely not recommended for production installation.

ETA is probably within a week or so - sorry for the continued delays.

**Comment by simonmichael** [comment]
Thanks.

**Comment by nestedsoftware** [comment]
Hi, any update on this?

**Comment by wohali** [comment]
Sorry, there is an administrative holdup - it is out of my hands.

**Comment by nestedsoftware** [comment]
Hi @pbgc @simonmichael - in the meantime, one option is to install couchdb 2.0 using the bitnami installer: https://bitnami.com/stack/couchdb/installer . After downloading the `.run` file, chmod +x it and then you can run that script to install it on your system.

**Comment by wohali** [comment]
@pbgc @simonmichael Let me see if I can push new 2.1.1 packages this weekend. I'll back up the current packages in case there is an issue.

**Comment by m-bryo** [comment]
We're waiting on you.

**Comment by wohali** [comment]
And I'm blocked by others on the CouchDB team.

Please be patient. I'm very sorry for the delay.

**Comment by rnewson** [comment]
@m-bryo please understand that we are all volunteers, you cannot make demands on our time. If you want to make Ubuntu packages, there's nothing stopping you doing it yourself. If you want to use whatever the couchdb team does instead, you will have to wait, patiently, for them to arrive.

**Comment by simonmichael** [comment]
I was a little disappointed to find that couchdb wasn't packaged in the latest ubuntu LTS. But to be clear "[darcs hub] ... went down down because of this" isn't correct.  It has been a contributing factor in my slowness in getting the site back up, but it wasn't the cause and it isn't a blocker now.

**Comment by pbgc** [comment]
@simonmichael When I said "because of this" never wanted to imply that it was apache/couchdb fault! Maybe it sounded like that but because I'm not a native speaker and could not express myself better. I have the greatest respect for open source projects work and it's volunteers!

**Comment by lichtamberg** [comment]
Any updates on this? We are also waiting for the release :)

**Comment by barbalex** [comment]
As disappointing it may be to us users that CouchDB is not yet available on Ubuntu 18.04: This is the kind of stuff that can really demotivate volunteers feeling responsible.

THANKS A LOT to you all: @wohali @rnewson and whoever I forgot to mention!

**Comment by m-bryo** [comment]
@rnewson not like I'm making  demands on your time as you put it. But ofcourse, I use couchdb for a reason... I understand you're all volunteers, but I was hoping there could be quicker releases. I've been stuck with the snap release (2.0.0)... It works, but a few things are differnet...

**Comment by m-bryo** [comment]
@rnewson as for the compiling from source... I'm not sure I'm willing to messup my system configs... If it was python or java... I could have done that. Erlang is just so alien for me.

**Comment by m-bryo** [comment]
@rnewson **apologies** though if I sounded too demanding.

**Comment by browndav** [comment]
Hi,

I've now seen a draft proposal to EOL CouchDB 1.7.x via e-mail. This doesn't sound like a spectacular idea to me when there's still limited availability of 2.x packages for major distros such as this one. Please consider maintaining v1.7.x security patches for the foreseeable future – people who depend upon distro packages and can't/won't build production dependencies themselves from source need sufficient time to evaluate, test, refactor, and migrate.

Thanks,
Dave

**Comment by wohali** [comment]
Thanks for the feedback @browndav . 

FYI, 1.x packages don't come from the Apache CouchDB project, only source code releases.

**Comment by browndav** [comment]
Thanks. FWIW, on the current CouchDB page I see a precompiled Windows distribution (x86), with OS X and Ubuntu packages labelled "TBD" – this may be contributing to confusion on my side.
